apps/language_server/lib/language_server/server.ex

```diff
@@ -789,4 +790,14 @@ defmodule ElixirLS.LanguageServer.Server do
   defp set_project_dir(state, _) do
     state
   end
+
+  defp create_gitignore(%{project_dir: project_dir} = state) do
+    with gitignore_path <- Path.join([project_dir, ".elixir_ls", ".gitignore"]),
+         :ok <- gitignore_path |> Path.dirname() |> File.mkdir_p(),
+         :ok <- File.write(gitignore_path, "*", [:read]) do
```

Comment:
Why use `:read` here? Wouldn't `:write` make more sense?

Reply:
@axelson My intention is `.gitignore` should not written by anyone except owner of this file. That's why I set it to `:read`.

Reply:
You're right. This should be `:write`